modin/backends/pandas/query_compiler.py

```diff
@@ -2837,7 +2834,11 @@ def compute_groupby(df, drop=False, partition_idx=0):
             except (ValueError, KeyError):
                 return compute_groupby(df.copy(), drop, partition_idx)
 
-        apply_indices = list(agg_func.keys()) if isinstance(agg_func, dict) else None
+        apply_indices = None
+        if isinstance(agg_func, dict):
+            apply_indices = tuple(agg_func.keys())
+        if selection is not None and apply_indices is None:
+            apply_indices = selection if is_list_like(selection) else (selection,)
```

Comment:
`apply_indices` is a parameter of `broadcast_apply_full_axis` which indicates the indices to apply a function on, so the function could be applied only to the required partitions